bofire/benchmarks/benchmark.py

```diff
@@ -2,23 +2,85 @@
 import os
 from abc import abstractmethod
 from copy import deepcopy
-from typing import Callable, List, Optional, Protocol, Tuple
+from typing import Callable, List, Literal, Optional, Protocol, Tuple, Union
 
 import numpy as np
 import pandas as pd
 from multiprocess.pool import Pool
+from pydantic import Field, PositiveFloat
+from scipy.stats import norm, t, uniform
 from tqdm import tqdm
+from typing_extensions import Annotated
 
 import bofire.strategies.api as strategies
+from bofire.data_models.base import BaseModel
 from bofire.data_models.domain.api import Domain
 from bofire.data_models.strategies.api import AnyStrategy
 
 
+class OutlierPrior(BaseModel):
+    type: str
+
+
+class UniformOutlierPrior(OutlierPrior):
+    type: Literal["UniformOutlierPrior"] = "UniformOutlierPrior"
+    bounds: Tuple[float, float]
+
+
+class NormalOutlierPrior(OutlierPrior):
+    type: Literal["NormalOutlierPrior"] = "NormalOutlierPrior"
+    loc: float
+    scale: PositiveFloat
+
+
+class tOutlierPrior(OutlierPrior):
+    type: Literal["tOutlierPrior"] = "tOutlierPrior"
+    loc: float
+    scale: PositiveFloat
+    df: PositiveFloat
+
+
+AnyOutlierPrior = Union[UniformOutlierPrior, NormalOutlierPrior, tOutlierPrior]
+
+
 class Benchmark:
+    def __init__(
+        self,
+        outlier_rate: Annotated[float, Field(ge=0, lt=1)] = 0,
+        OutlierPrior: Optional[AnyOutlierPrior] = None,
+    ):
+        self.outlier_rate = outlier_rate
+        self.OutlierPrior = OutlierPrior
+
     def f(
-        self, candidates: pd.DataFrame, return_complete: bool = False
+        self,
+        candidates: pd.DataFrame,
+        return_complete: bool = False,
     ) -> pd.DataFrame:
         Y = self._f(candidates)
+        if self.OutlierPrior is not None:
+            # no_outliers = int(len(Y) * self.outlier_rate)
+            ix2 = np.zeros(len(Y), dtype=bool)
+            ix1 = uniform().rvs(len(Y))
+            # ix2[np.random.choice(len(Y), no_outliers, replace=False)] = True
+            ix2 = ix1 <= self.outlier_rate
+            no_outliers = sum(ix2)
+
+            if self.OutlierPrior.type == "UniformOutlierPrior":
+                Y.loc[ix2, "y"] = Y.loc[ix2, "y"] + uniform(
+                    self.OutlierPrior.bounds[0],
+                    self.OutlierPrior.bounds[1] - self.OutlierPrior.bounds[0],
+                ).rvs(
+                    no_outliers
+                )  # type: ignore
+            if self.OutlierPrior.type == "NormalOutlierPrior":
+                Y.loc[ix2, "y"] = Y.loc[ix2, "y"] + norm(
+                    self.OutlierPrior.loc, self.OutlierPrior.scale
+                ).rvs(no_outliers)
+            if self.OutlierPrior.type == "tOutlierPrior":
```

Comment:
No need for `tOutlierPrior`. Uniform and Normal should be sufficient. Please also move the sampling of the outliers into the specific classes above.